### PR TITLE
GFORMS-988 Added ability to inhibit section title at page level and s…

### PR DESCRIPTION
**Pull request by girishkamat**
…ection instruction level

### CODE REVIEW

File: app/uk/gov/hmrc/gform/summary/SummaryRenderingService.scala [hunk 1]
```diff
@@ -366,9 +364,13 @@ object SummaryRenderingService {
...
+          .filter(_ == InvisiblePageTitle)
+          .fold(begin_section(pageTitle))(_ => HtmlFormat.empty)
       } { _ =>
-        begin_section(pageTitle)
+        page.presentationHint
+          .filter(_ == InvisiblePageTitle)
+          .fold(begin_section(pageTitle))(_ => HtmlFormat.empty)
       } { addToList =>
         addToList.presentationHint
           .filter(_ == InvisiblePageTitleInSummary)
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/summary/SummaryRenderingService.scala:376]: What is the difference between `InvisiblePageTitleInSummary` and `InvisiblePageTitle` ?

File: app/uk/gov/hmrc/gform/gform/FormComponentUpdater.scala [hunk 2]
```diff
@@ -75,7 +75,11 @@ class FormComponentUpdater(formComponent: FormComponent, index: Int, baseIds: Li
     shortName = formComponent.shortName.map(expandSmartString),
     errorMessage = formComponent.errorMessage.map(expandSmartString),
     validators = formComponent.validators.map(expandFormComponentValidator),
-    instruction = formComponent.instruction.map(i => i.copy(name = expandSmartString(i.name)))
+    instruction = formComponent.instruction.map(i =>
+      i.name match {
+        case Some(name) => i.copy(name = Some(expandSmartString(name)))
+        case _          => i
+    })
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/gform/FormComponentUpdater.scala:82]: What about to do rather:

```suggestion
    instruction = formComponent.instruction.map(i => i.copy(name = i.name.map(expandSmartString)))
```

File: app/uk/gov/hmrc/gform/gform/PageUpdater.scala [hunk 3]
```diff
@@ -30,20 +30,24 @@ class PageUpdater[A <: PageMode](page: Page[A], index: Int, baseIds: List[FormCo
...
+      fields = page.fields.map { field =>
+        new FormComponentUpdater(field, index, baseIds).updatedWithId
+      },
+      validators = page.validators.map(expandValidator),
+      continueLabel = page.continueLabel.map(expandSmartString),
+      instruction = page.instruction.map(i =>
+        i.name match {
+          case Some(name) => i.copy(name = Some(expandSmartString(name)))
+          case _          => i
+      })
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/gform/PageUpdater.scala:49]: Similar to above, what about to do it one-liner?

File: app/uk/gov/hmrc/gform/models/FormModelExpander.scala [hunk 4]
```diff
@@ -129,7 +129,11 @@ object FormModelExpander {
         fields = page.fields.map { field =>
           new FormComponentUpdater(field, index, source.allIds).updatedWithId
         },
-        instruction = page.instruction.map(i => i.copy(expand(i.name)))
+        instruction = page.instruction.map(i =>
+          i.name match {
+            case Some(name) => i.copy(name = Some(expand(name)))
+            case _          => i
+        })
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/models/FormModelExpander.scala:136]: Yet another one-liner perhaps?

**Review by josef-vlach — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/gform/summary/SummaryRenderingService.scala (hunk 1 above)
Comment (girishkamat) [reply app/uk/gov/hmrc/gform/summary/SummaryRenderingService.scala:376]: `InvisiblePageTitleInSummary` was added to hide add to list title in summary page. I know the name doesnt do justice to what it does. `InvisiblePageTitle` applies to page titles, for all pages - a generic feature to disable titles

File: app/uk/gov/hmrc/gform/gform/FormComponentUpdater.scala (hunk 2 above)
Comment (girishkamat) [reply app/uk/gov/hmrc/gform/gform/FormComponentUpdater.scala]: good suggestion, done

File: app/uk/gov/hmrc/gform/gform/PageUpdater.scala (hunk 3 above)
Comment (girishkamat) [reply app/uk/gov/hmrc/gform/gform/PageUpdater.scala]: done

File: app/uk/gov/hmrc/gform/models/FormModelExpander.scala (hunk 4 above)
Comment (girishkamat) [reply app/uk/gov/hmrc/gform/models/FormModelExpander.scala]: done

**Review by josef-vlach — APPROVED** [review]

**Pull request merged**